benchadapt/python/benchadapt/adapters/_adapter.py

```diff
@@ -89,7 +90,10 @@ def transform_results(self) -> List[BenchmarkResult]:
         """
         log.info("Transforming results for conbench")
         results = self._transform_results()
-        self.results = [self.update_benchmark_result(res) for res in results]
+        run_id = uuid.uuid4().hex
```

Comment:
Will this generate a new runid for each json file that is parsed? (this might be true, I haven't followed it in detail)

Here's an example of this happening in velox: https://github.com/facebookincubator/velox/blob/a5821449193254916293bf509faf7ec66ae6a9be/scripts/veloxbench/veloxbench/implemented_benchmarks.py#L34-L44

For those, each of those json files we want to be in a single run.

Reply:
No, unless you set things up weirdly so you create a new adapter instance for each file and don't specify the `run_id` elsewhere. That's a bad idea; an adapter instance should always correspond to a run, and the iteration over files should happen in the `_transform_results()` method of the adapter, e.g. [here](https://github.com/conbench/conbench/blob/main/benchadapt/python/benchadapt/adapters/folly.py#L51) for folly.

Reply:
Cool, when we do get to documenting all of this in a more user-facing way, let's make sure to add "create a new adapter instance for each file and don't specify the run_id elsewhere. That's a bad idea" to that

Reply:
I added more to the docs both for the abstract adapter and the result now because it's all fresh in my brain